crates/runtime/src/datafusion/query/error_code.rs

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use datafusion::error::DataFusionError;
+
+pub enum ErrorCode {
+    SyntaxError,
+    QueryPlanningError,
+    DataReadError,
+    UnexpectedError,
+}
+
+impl std::fmt::Display for ErrorCode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ErrorCode::SyntaxError => write!(f, "SyntaxError"),
+            ErrorCode::QueryPlanningError => write!(f, "QueryPlanningError"),
+            ErrorCode::DataReadError => write!(f, "DataReadError"),
+            ErrorCode::UnexpectedError => write!(f, "UnexpectedError"),
+        }
+    }
+}
+
+impl From<&ErrorCode> for i8 {
```

Comment:
This is used to make `execution_status` result more granular in case of failure - specific code instead of generic -1